pkgs/development/python-modules/libcst/default.nix

```diff
@@ -1,42 +1,63 @@
-{ lib, buildPythonPackage, fetchFromGitHub, pythonOlder, black, isort
-, pytestCheckHook, pyyaml, typing-extensions, typing-inspect, dataclasses }:
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, pythonOlder
+, black
+, isort
+, pytestCheckHook
+, pyyaml
+, typing-inspect
+, dataclasses
+, hypothesis
+}:
 
 buildPythonPackage rec {
   pname = "libcst";
-  version = "0.3.13";
+  version = "0.3.17";
+  disabled = pythonOlder "3.6";
 
   # Some files for tests missing from PyPi
   # https://github.com/Instagram/LibCST/issues/331
   src = fetchFromGitHub {
     owner = "instagram";
     repo = pname;
     rev = "v${version}";
-    sha256 = "0pbddjrsqj641mr6zijk2phfn15dampbx268zcws4bhhhnrxlj65";
+    sha256 = "sha256-mlSeB9OjCiUVYwcPYNrQdlfcj9DV/+wqVWt91uFsQsU=";
   };
 
-  disabled = pythonOlder "3.6";
+  propagatedBuildInputs = [
+    pyyaml
+    typing-inspect
+  ] ++ lib.optional (pythonOlder "3.7") dataclasses;
 
-  propagatedBuildInputs = [ pyyaml typing-inspect ]
-    ++ lib.optional (pythonOlder "3.7") dataclasses;
+  checkInputs = [
+    black
+    isort
+    pytestCheckHook
+    hypothesis
+  ];
 
-  checkInputs = [ black isort pytestCheckHook ];
+  postPatch = ''
+    # Circular dependency hypothesmith, https://github.com/Instagram/LibCST/issues/346
+    substituteInPlace requirements-dev.txt --replace "hypothesmith>=0.0.4" ""
```

Comment:
This is not required because hypothesmith is build without libcst tests. https://github.com/NixOS/nixpkgs/blob/a72f92cc89582f89d9f71ebaef04ae4bca47f3af/pkgs/top-level/python-packages.nix#L3113

Reply:
#114452 update libcst to 0.3.17. Thus, this PR can be compressed to two commit.